修复 field crypto 组件 处理 mp wrapper 时 无法加密的情况

hxlcw opened the pull request:
<!--
  非常感谢您的 PR！在提交之前，请务必确保您 PR 的代码经过了完整测试，并且通过了代码规范检查。
-->

<!-- 在 [] 中输入 x 来勾选) -->

## PR 类型

<!-- 您的 PR 引入了哪种类型的变更？ -->
<!-- 只支持选择一种类型，如果有多种类型，可以在更新日志中增加 “类型” 列。 -->

- [ ] 新 feature
- [x] Bug 修复
- [ ] 功能增强
- [ ] 文档变更
- [ ] 代码样式变更
- [ ] 重构
- [ ] 性能改进
- [ ] 单元测试
- [ ] CI/CD
- [ ] 其他

## PR 目的

修复 field crypto 组件 处理 mp wrapper 时 无法加密的情况

## 解决方案

通过在mybatis 拦截器中获取参数中的`ew` 即`wrapper` 参数，对 `wrapper` 参数进行解析，并通过`mapper` 获取泛型DO对象，通过匹配`wrapper` 的参数 和 DO 对象中的字段注解，如果`wrapper` 中有使用带有`@FieldEncrypt`的字段，则将其加密 并 替换 原有的值。

## PR 测试

1. 项目中有测试用例
- 通过 `getById` 测试解密 和 加密 后的值是否匹配
- 通过 `mqUdate` 测试 lambda + wrapper 的情况下 加密是否起效

2. 通过 continew-admin 配置 systemPath 的方式 连接本地的 crypto 包，测试系统是否成功运行。

## Changelog

| 模块  | Changelog | Related issues |
|-----|-----------| -------------- |
|   security-crypto  |      修复字段加密在处理 mp wrapper 情况时失效的问题  |                |
|   security-crypto  |      添加测试用例 |                |

<!-- 如果有多种类型的变更，可以在变更日志表中增加 “类型” 列，该列的值与上方 “PR 类型” 相同。 -->
<!-- Related issues 格式为 Closes #<issue号>，或者 Fixes #<issue号>，或者 Resolves #<issue号>。 -->

## 其他信息

<!-- 请描述一下还有哪些注意事项。例如：如果引入了一个不向下兼容的变更，请描述其影响。 -->

## 提交前确认

- [x] PR 代码经过了完整测试，并且通过了代码规范检查
- [x] 已经完整填写 Changelog，并链接到了相关 issues
- [x] PR 代码将要提交到 dev 分支

hxlcw committed:
fix(security-crypto): 修复字段加密在处理 mp wrapper 情况时失效的问题

hxlcw committed:
test(security-crypto): 添加【getById、mpUpdate】测试方法

hxlcw committed:
fix(security-crypto): 修复获取 mapper 泛型的问题

hxlcw committed:
test(security-crypto): 规范化测试用例，使用断言判断测试结果

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/Charles7c/continew-starter?pullRequest=4) <br/>All committers have signed the CLA.

hxlcw committed:
chore(security-crypto): 移除测试用例和POM中的测试依赖

hxlcw committed:
fix(security-crypto): 避免重新赋值方法参数，并解决多接口继承时获取Domain的问题
- 修改了 `getDoByMapperClass` 方法，通过引入新的局部变量来避免对参数的重新赋值
- 优化了方法逻辑，确保在多接口继承的情况下能够正确获取Domain类型
